src/test/ui/consts/missing_span_in_backtrace.stderr

```diff
@@ -0,0 +1,23 @@
+error[E0080]: evaluation of constant value failed
+/rustc/xyz/library/core/src/ptr/mod.rs:925:14: inside `swap_nonoverlapping::<MaybeUninit<u8>>`
+/rustc/xyz/library/core/src/ptr/mod.rs:944:9: inside `ptr::swap_nonoverlapping_simple_untyped::<MaybeUninit<u8>>`
+--> /rustc/xyz/library/core/src/ptr/mod.rs:1135:9
+note: unable to copy parts of a pointer from memory at alloc10
+note: inside `std::ptr::read::<MaybeUninit<MaybeUninit<u8>>>`
+/rustc/xyz/library/core/src/mem/mod.rs:773:17: inside `mem::swap_simple::<MaybeUninit<MaybeUninit<u8>>>`
+   |
+  ::: $DIR/missing_span_in_backtrace.rs:16:9
```

Comment:
I think the expected output for these (which doesn't *need* to be followed, but I'd prefer to) would be

```suggestion
error[E0080]: evaluation of constant value failed
  |
  = /rustc/xyz/library/core/src/ptr/mod.rs:925:14: inside `swap_nonoverlapping::<MaybeUninit<u8>>`
  = /rustc/xyz/library/core/src/ptr/mod.rs:944:9: inside `ptr::swap_nonoverlapping_simple_untyped::<MaybeUninit<u8>>`
 --> /rustc/xyz/library/core/src/ptr/mod.rs:1135:9
  |
  = note: unable to copy parts of a pointer from memory at alloc10
  = note: inside `std::ptr::read::<MaybeUninit<MaybeUninit<u8>>>`
  = /rustc/xyz/library/core/src/mem/mod.rs:773:17: inside `mem::swap_simple::<MaybeUninit<MaybeUninit<u8>>>`
   |
  ::: $DIR/missing_span_in_backtrace.rs:16:9
```

Aligning with the available path (so accounting for the biggest number shown in the gutter) and separating the message from the notes given no available span.

Reply:
Also, what about the following formatting?

```suggestion
error[E0080]: evaluation of constant value failed
  --> /rustc/xyz/library/core/src/ptr/mod.rs:1135:9
   |
note: inside `swap_nonoverlapping::<MaybeUninit<u8>>`
  --> /rustc/xyz/library/core/src/ptr/mod.rs:925:14
   |
note: inside `ptr::swap_nonoverlapping_simple_untyped::<MaybeUninit<u8>>`
  --> /rustc/xyz/library/core/src/ptr/mod.rs:944:9
   |
   = note: unable to copy parts of a pointer from memory at alloc10
   = note: inside `std::ptr::read::<MaybeUninit<MaybeUninit<u8>>>`
   = /rustc/xyz/library/core/src/mem/mod.rs:773:17: inside `mem::swap_simple::<MaybeUninit<MaybeUninit<u8>>>`
   |
  ::: $DIR/missing_span_in_backtrace.rs:16:9
```

I also noticed that the new pseudo-notes come before the path for the main diagnostic, that should be changed.

Reply:
One last thing: if there are span labels for the primary span, they should be displayed first and as a `= note:` without a path. I think that with all of these we stay inside the established grammar of the printed diagnostics.

Reply:
At the emitter stage we do not have enough information to figure this out. Labels are reordered to fit the file structure. I will check if converting them to notes at const eval error time will still yield reasonable diagnostics